[WIP]: IAM policy with permissions required by installer

alexsomesan opened the pull request:
This is the role policy I was able to use to successfully apply / destroy a cluster.

This is WIP as I need to add documentation on how to use this policy.

/cc: @sym3tri

s-urbaniak commented on examples/installer-iam-policy.json:
```diff
@@ -0,0 +1,169 @@
+            "Resource": "*"
+        }
+    ],
+    "Version": "2012-10-17"
```
silly question: what is this magic date about?

robszumski replied:
Amazon's terrible versioning scheme...

alexsomesan replied:
What Rob said.
It's the date that particular IAM API version was released.

alexsomesan commented:
I will update that one as well, or better yet we can have the website point to this one.

I would prefer to have the reference copy in here so I can reference it in instructions on how to use and also have CLI examples or a Terraform template that users can run to create the said role.

@sym3tri What do you think?

sym3tri commented:
@alexsomesan I'd prefer we only keep 1 file to ease maintenance. 

Docs can't point to this one b/c all website docs are rooted in `/Documentation` and can't reference files not in the dir.

Having the examples reference files in `Documentation` or the website URL seems easiest to me.

alexsomesan commented:
I agree on the 1 file approach.

How about if we move this under /Documentation?


On Thu 18. May 2017 at 21:04, Ed Rooth <notifications@github.com> wrote:

> @alexsomesan <https://github.com/alexsomesan> I'd prefer we only keep 1
> file to ease maintenance.
>
> Docs can't point to this one b/c all website docs are rooted in
> /Documentation and can't reference files not in the dir.
>
> Having the examples reference files in Documentation or the website URL
> seems easiest to me.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/coreos/tectonic-installer/pull/756#issuecomment-302511562>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABC-m1ueiAEOXTYF5NF2icTCaFHGWEH4ks5r7JZHgaJpZM4Nfjfk>
> .
>

robszumski commented:
Can we make sure this lands in the same spot as the old example? Same filename, etc? This way we don't have to update any links?

sym3tri commented:
`/Documentation/files/aws-policy.json`

sym3tri commented:
@alexsomesan status on this? Needs to merge before we can release.

alexsomesan commented:
Pending writing usage instructions.
I'll update before EOD today.

On Tue 23. May 2017 at 20:40, Ed Rooth <notifications@github.com> wrote:

> @alexsomesan <https://github.com/alexsomesan> status on this? Needs to
> merge before we can release.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/coreos/tectonic-installer/pull/756#issuecomment-303493492>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABC-m20Fm7pqIexImuXLke7hmPzVOaJ2ks5r8yf-gaJpZM4Nfjfk>
> .
>

sym3tri commented:
Thanks. I believe we can rely on the existing usage instructions in our docs.
https://coreos.com/tectonic/docs/latest/install/aws/requirements.html#temporary-credentials

alexsomesan commented:
@sym3tri 
* moved the policy in the right location
* updated the example assume role policy document to reference users rather than a service
* included instructions for using the role directly from terraform

sym3tri commented:
@alexsomesan safe to remove the `do-not-merge` label and merge?

sym3tri commented:
@alexsomesan I'm going to merge this for now since it's a release blocker and LGTM and is better than what we have in master today. Please amend later if needed.

alexsomesan commented:
Sorry for forgetting the label on this. It was actually good to merge from my POV.
Thanks for taking care of that!
